contracts/v2/PolygonZkEVMBridgeV2.sol

```diff
@@ -726,6 +734,33 @@ contract PolygonZkEVMBridgeV2 is
             ];
     }
 
+    /**
+     * @notice Set the address of a wrapper using the token information if already exist
+     * @dev This function is used to allow any existing token to be mapped with
+     *      origin token. Wrapper contract should handle mint/burn of the existing token.
+     * @param originNetwork Origin network
+     * @param originTokenAddress Origin token address, 0 address is reserved for ether
+     * @param wrappedTokenAddress Arbitary contract that implements TokenWrapped interface
+     */
+    function setTokenWrappedAddress(
+        uint32 originNetwork,
+        address originTokenAddress,
+        address wrappedTokenAddress,
+        address existingTokenAddress
+    ) external onlyRollupManager {
+        // Handle claimAsset on target chain
+        bytes32 tokenInfoHash = keccak256(
+            abi.encodePacked(originNetwork, originTokenAddress)
+        );
+        tokenInfoToWrappedToken[tokenInfoHash] = wrappedTokenAddress;
+
+        // Handle bridgeAsset from origin chain
+        wrappedTokenToTokenInfo[
+            existingTokenAddress
+        ] = TokenInformation(originNetwork, originTokenAddress);
```

Comment:
if originTokenAddress == address(0), this will make bridgeAsset think that existingTokenAddress is a "native" token from the current chain

Reply:
> this will make bridgeAsset think that existingTokenAddress is a "native" token from the current chain

Sorry ser, can you give me a link to lines where `bridgeAsset` will think token is native when `originTokenAddres` is zero addy?

AFAIK, `bridgeAsset` will think token address is native when `token` address itself is zero